### PR TITLE
update models

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

**Review by Cheephot — APPROVED** [review]

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 1]
```diff
@@ -48,57 +46,66 @@ public class CreateUserCommand : ICreateUserCommand
 
     #region private methods
 
-    private void ChangeUserDepartment(Guid departmentId, Guid userId, List<string> errors)
+    private void EditCompanyEmployee(Guid? departmentId, Guid? positionId, Guid? officeId, Guid userId, List<string> errors)
     {
-      string errorMessage = $"Can't assign user {userId} to the department {departmentId}. Please try again later.";
-      const string logMessage = "Can't assign user {UserId} to the department {DepartmentId}.";
+      string departmentErrorMessage = $"Cannot assign position to user. Please try again later.";
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:51]: maybe cannot assign department......?

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 2]
```diff
@@ -48,57 +46,66 @@ public class CreateUserCommand : ICreateUserCommand
...
+      string positionErrorMessage = $"Cannot assign department to user. Please try again later.";
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:52]: position?

File: src/UserService.Business/Commands/User/CreateUserCommand.cs [hunk 3]
```diff
@@ -48,57 +46,66 @@ public class CreateUserCommand : ICreateUserCommand
...
+
+        if (!response.IsSuccess)
         {
-          _logger.LogWarning(logMessage, userId, departmentId);
+          _logger.LogWarning(logMessage, userId);
 
-          errors.Add(errorMessage);
+          errors.Add(departmentErrorMessage);
+          errors.Add(positionErrorMessage);
+          errors.Add(officeErrorMessage);
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/User/CreateUserCommand.cs:76]: maybe it better if you add the only message of department/position/office if request the entity != null?
if(department != null) {errors.Add(departmentErrorMessage)}.......

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 4]
```diff
@@ -22,337 +19,327 @@
...
-            IRequestClient<IChangeUserRoleRequest> rcRole,
-            IRequestClient<IChangeUserOfficeRequest> rcOffice,
-            IHttpContextAccessor httpContextAccessor,
-            IBus bus)
+      if (avatarRequest == null)
+      {
+        return null;
+      }
+
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:146]: maybe it is better if we create global variable with requester user Id? we use it in all private methods and every time pull it.

File: src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 5]
```diff
@@ -11,65 +11,70 @@
...
+    // search
 
-        [AutoInjectRequest(typeof(ISearchUsersRequest))]
-        public string SearchUsersEndpoint { get; set; }
+    [AutoInjectRequest(typeof(ISearchUsersRequest))]
+    public string SearchUsersEndpoint { get; set; }
 
-        [AutoInjectRequest(typeof(IChangeUserRoleRequest))]
-        public string ChangeUserRoleEndpoint { get; set; }
+    // rights
```
Comment (MariyaSin) [inline src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs:52]: do we need this comment?

File: src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 6]
```diff
@@ -11,65 +11,70 @@
...
-    }
+    [AutoInjectRequest(typeof(IGetPositionsRequest))]
+    public string GetPositionsEndpoint { get; set; }
+
+    [AutoInjectRequest(typeof(IEditCompanyEmployeeRequest))]
+    public string EditCompanyEmployeeEndpoint { get; set; }
+
+    [AutoInjectRequest(typeof(IGetCompanyEmployeesRequest))]
+    public string GetCompanyEmployeesEndpoint { get; set; }
+
```
Comment (MariyaSin) [inline src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs:73]: remove line

File: src/UserService.Models.Dto/Requests/User/EditUserRequest.cs [hunk 7]
```diff
@@ -5,7 +5,7 @@ namespace LT.DigitalOffice.UserService.Models.Dto.Requests.User
 {
     public class EditUserRequest
     {
-        public Guid DepartmentId { get; set; }
+        public Guid? DepartmentId { get; set; }
         public Guid PositionId { get; set; }
         public Guid RoleId { get; set; }
```
Comment (MariyaSin) [inline src/UserService.Models.Dto/Requests/User/EditUserRequest.cs:10]: user can not have role

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:146]: I don't think so, it is not property of class in meaning of logic, i don't know how to say better

File: src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs:52]: this config is big, so I decide to add this comment for creating structure of config

Comment (BelehovEgor) [reply src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs:52]: also it is not safety

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: fixed

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: fixed

File: src/UserService.Business/Commands/User/CreateUserCommand.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/CreateUserCommand.cs]: fixed

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:146]: also i think it is unsafety

File: src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs (hunk 6 above)
Comment (BelehovEgor) [reply src/UserService.Models.Dto/Configurations/RabbitMqConfig.cs:73]: removed

File: src/UserService.Models.Dto/Requests/User/EditUserRequest.cs (hunk 7 above)
Comment (BelehovEgor) [reply src/UserService.Models.Dto/Requests/User/EditUserRequest.cs:10]: why

**Review by MariyaSin — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**